### PR TITLE
Add simplified query API

### PR DESCRIPTION
**Pull request by chernser**
## Summary
Query API returns response object with bare InputStream. User have to choose what reader to use for reading the stream. 
Extended API makes things a bit simpler because it requests data in descriptive format and creates a reader upfront

### CODE REVIEW

File: client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java [hunk 1]
```diff
@@ -103,6 +104,26 @@ public void testSimpleQueryWithTSV() {
...
+    }
+
+    @Test(groups = {"integration"})
+    public void testReadQuery() {
+        prepareSimpleDataSet();
+
+        Future<QueryResponseReader> response = client.readQuery("SELECT * FROM " + DATASET_TABLE);
+
+        try {
+            ClickHouseBinaryFormatReader reader = response.get().getReader();
```
Comment (mzitnik) [inline client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java:117]: why not have a records method?

File: client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java [hunk 2]
```diff
@@ -103,6 +104,26 @@ public void testSimpleQueryWithTSV() {
...
+            Map<String, Object> record = new HashMap<>();
+            while (reader.next()) {
+                reader.copyRecord(record);
```
Comment (mzitnik) [inline client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java:120]: Do we have examples with getString / getLong / ... methods

**Review by mzitnik — APPROVED** [review]

File: client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java (hunk 2 above)
Comment (chernser) [reply client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java:120]: yes in the new example `com.clickhouse.examples.client_v2.SimpleReader`

File: client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java (hunk 1 above)
Comment (chernser) [reply client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java:117]: Agree. I will think and implement it more natural way as we discussed.

**Review by mzitnik — APPROVED** [review]

**Review by mzitnik — APPROVED** [review]

**Pull request merged**